iconservice/icon_service_engine.py

```diff
@@ -441,6 +444,105 @@ def _invoke_request(self,
+            context.step_counter.apply_step(StepType.DEFAULT, 1)
+
+            input_size = self._get_byte_length(params.get('data', None))
+            context.step_counter.apply_step(StepType.INPUT, input_size)
+
+        if data_type == "deploy":
+            data_size = self._get_byte_length(data.get('content', None))
+            context.step_counter.apply_step(StepType.CONTRACT_SET, data_size)
+            # When installing SCORE.
+            if to == ZERO_SCORE_ADDRESS:
+                context.step_counter.apply_step(StepType.CONTRACT_CREATE, 1)
+            # When updating SCORE.
+            else:
+                context.step_counter.apply_step(StepType.CONTRACT_UPDATE, 1)
+
+    def _estimate_step_by_execution(self, request, context, step_limit):
+        """Processes the transaction and estimates step.
+
+        :param request:
+        :param context:
+        :param step_limit:
+        """
+        # Processes and estimates steps without commit.
+        method: str = request.get('method')
+        params: dict = request.get('params')
+        data_type: str = params.get('dataType')
+        from_: Address = params.get('from')
+        to: Address = params.get('to')
+
+        if data_type == "deploy":
+            return
+
+        if data_type == "call" or (not data_type == "call" and to.is_contract):
+            timestamp = params.get('timestamp', self._icx_storage.last_block.timestamp)
+            context.tx = Transaction(origin=from_,
+                                     index=0,
+                                     tx_hash=sha3_256(str(timestamp).encode()),
+                                     timestamp=timestamp,
+                                     nonce=params.get('nonce', None))
+            context.msg = Message(sender=from_, value=params.get('value', 0))
+            context.current_address = to
+            context.event_logs: List['EventLog'] = []
+            context.logs_bloom: BloomFilter = BloomFilter()
+            context.traces: List['Trace'] = []
+
+            context.block = self._precommit_data_manager.last_block
+            context.block_batch = BlockBatch(Block.from_block(context.block))
+            context.tx_batch = TransactionBatch()
+            context.new_icon_score_mapper = IconScoreMapper()
+
+            # Deposits virtual ICXs to the sender to prevent validation error due to 'out of balance'.
+            account = self._icx_storage.get_account(context, from_)
+            account.deposit(step_limit * self._get_step_price() + params.get('value', 0))
+            self._icx_storage.put_account(context, from_, account)
+            self._call(context, method, params)
+
+    def estimate_step(self, request: dict) -> int:
+        """
+        Estimates the amount of step to process a specific transaction.
+
+        [Specific information of each step]
+        1. Transfer coin
+            1) When where to send coin is EOA: Default
```

Comment:
I'd suggest "When the destination is EOA", and the same for the lines below

Reply:
Thanks for the good suggestion. I'll reflect it.